### PR TITLE
Fix: simplify tests CI structure

### PR DESCRIPTION
**Pull request by 12rambau**
We had prior to this PR 2 CI github actions so I decided to wrap everything in the `unit.yaml` one. It required more modifications than I expected. 

### unit.yaml 

In this file I added the `mypy` test and run it prior to any other CI. thus when the action starts it checks for `mypy` and `lint` and if they don't succeed it doesn't even start the `test` matrix.

I adapted the `lint` action to avoid the warning about `node` from Github side 
`mypy` checks are done through a `nox` session. 

Tests are run against the same matrix as the previous CI the 3 only changes are: 
- I don't reinstall python3 as it's already there 
- The test are run through a `nox` session 
- I export for version 3.10 the result of the coverage test in xml format

Once everything is done the result of the coverage is sent to codecov so we keep track of how the project evolves. 

### noxfile.py 

I created sessions for multiple step: 

- `ci_test`: the pytest tests run in the CI exporting the coverage report as XML. 
- `test`: the pytest tests run in local builds. They are not yet working. I think they should integrate some default database build (sqllite ?) to have something to check against. They would allow to run the tests locally before flooding the Github actions.
- `mypy`: as its name suggest it runs mypy in a dedicated session. 

### misc 

- In the process I changed the badges so they connect to the new workflows (and link to appropriate pages) 
- I authorized the codecov service on the repository, it will not be activated until it's merged to default
- added few dependencies that were missing for the tests 
- `ruff` has a new "lint" sub group so I listened to their warning and adapted the instructions

## next step 

In a follow-up PR I would like to make the `test` session to work locally.

### CODE REVIEW

File: .github/workflows/unit.yaml [hunk 1]
```diff
@@ -11,8 +11,78 @@ jobs:
...
+      - name: Setup database schema
+        run: psql -h localhost -U eralchemy -c 'CREATE SCHEMA eralchemy_test;'
+        env:
+          PGPASSWORD: eralchemy
+      - name: Install nox
+        run: pip install nox
+      - name: Test with pytest
+        run: nox -s ci_test
+      - uses: actions/upload-artifact@v4
+        if: ${{ matrix.python-version == '3.10' }}
```
Comment (maurerle) [inline .github/workflows/unit.yaml:71]: use 3.12 here?

**Review by maurerle — CHANGES_REQUESTED** [review]
We would be going from "we are testing different sqlalchemy versions and python combination" to only test the python versions right?
I would like to still keep testing the different sqlalchemy version?

File: .github/workflows/unit.yaml (hunk 1 above)
Comment (12rambau) [reply .github/workflows/unit.yaml:71]: my go to is to take the default version of ubuntu latest, I can check if it's not 3.10. In any case i'm sure it's not 3.12

**Comment by 12rambau** [comment]
@maurerle what versions of sqlalchemy should we try and in combination with which versions of python ? 
I think the default should go for latest and only specific ones should go to specific SA, like oldest python with oldest sqlalchemy

**Comment by maurerle** [comment]
@12rambau yes sounds good! I would like to test python 3.8, 3.10 and 3.12, each with SQLAlchemy 1.4 and with latest SQLAlchemy..?
Or just 3.8 with 1.4 and the other two with latest sqlalchemy

**Comment by 12rambau** [comment]
ok I'll keep 1.4 with Python 3.8 (that will be our bottom line and latest with the others (that's easier to setup).

**Comment by 12rambau** [comment]
The addition of pygraphviz is making everything to crash on the CI side, it's very enoying, any reason why we use both pygraphivz and graphiz python binding ?

**Comment by maurerle** [comment]
I don't see where pygraphviz is creating an error?
In the past there were people who would rather use either library due to ease of installation, so I kept the option to have either installed.
See #121 

In the current main branch, the CI works - if you have an issue, just point me there

**Comment by 12rambau** [comment]
@maurerle when I use a devcontainer I never managed to launch the nox test session from the terminal because of C/C++ error. That's not very important for now. Do you have any idea why the CI is not starting since my last commit in here ?

**Comment by maurerle** [comment]
Oh okay, I would have to look into this, I am not using the devcontainer for now.

Sure, see `The workflow is not valid. .github/workflows/unit.yaml (Line: 41, Col: 11): A mapping was not expected` from https://github.com/eralchemy/eralchemy/actions/runs/10899203644

**Comment by 12rambau** [comment]
@maurerle confirming there is a bug with sqlalchemy 1.4 see the logs for more information: https://github.com/eralchemy/eralchemy/actions/runs/10899722434/job/30245768185?pr=133

Can you advice on what should be done ? I would be in favor to upgrade the version of sqlalchemy to the first occurence of the missing module

**Comment by maurerle** [comment]
It now works with `sqlalchemy==1.4.18` and `flask-sqlalchemy<3.1`. I hope the flask-sqlalchemy version is solved automatically correctly

**Review by maurerle — APPROVED** [review]

**Pull request merged**